### PR TITLE
Sequential event fecth

### PR DESCRIPTION
**Pull request by AugustoL**
Get trade events form last 50k blocks, make request for every 5k blocks, if it reaches the trade event limit stop fetching, it ti reaches the block limit it stops searching too.

### CODE REVIEW

**Comment by pimato** [comment]
does this loads events all the time?

**Comment by AugustoL** [comment]
When it is connected yes, it fetch in block ranges of 5000 blocks till it reaches the maximun events, for example if you want to get 10 last events, it gets 7 form the last 5000 blocks, 1 the next 5000 and the remaining 3 the next 5000. got it?

**Comment by pimato** [comment]
why not add fetching only if the user scrolled reached the bottom at the trade history?

File: src/stores/datStore.ts [hunk 1]
```diff
@@ -84,7 +84,8 @@ export interface SellReturn {
 
 export type TradeEvent = BuyEvent | SellEvent;
 
-export const TRADES_FROM_LAST_BLOCKS = 1000;
+export const TRADES_FROM_LAST_BLOCKS = 50000;
```
Comment (jpkcambridge) [inline src/stores/datStore.ts:87]: I would suggest `MAX_BLOCKS_FOR_FETCHING_TRADES`

File: src/stores/datStore.ts [hunk 2]
```diff
@@ -84,7 +84,8 @@ export interface SellReturn {
...
+export const BLOCK_PER_TRADES_FETCH = 5000;
```
Comment (jpkcambridge) [inline src/stores/datStore.ts:88]: I think should be `BLOCKS_PER_TRADES_FETCH`

File: src/stores/datStore.ts [hunk 3]
```diff
@@ -407,19 +410,50 @@ export default class DatStore {
...
-        let combinedTrades: any[] = buyEvents.concat(sellEvents);
-        combinedTrades = combinedTrades.sort(function (a, b) {
-            return b.blockNumber - a.blockNumber;
+        let blockLimit = (this.rootStore.configStore.getDATinfo()).fromBlock;
+        let tradesToReturn = [];
+        
+        if (toBlock > BLOCK_PER_TRADES_FETCH) {
+          blockLimit = toBlock - TRADES_FROM_LAST_BLOCKS;     
+        }
+        if (toBlock > BLOCK_PER_TRADES_FETCH) {
```
Comment (jpkcambridge) [inline src/stores/datStore.ts:421]: Can these two conditions be combined into one?

Comment (AugustoL) [reply src/stores/datStore.ts:421]: they should, yes!

File: src/stores/datStore.ts [hunk 4]
```diff
@@ -407,19 +410,48 @@ export default class DatStore {
...
-        const sellEvents = await this.fetchSellEvents(datAddress, numToGet, fromBlock);
-
-        let combinedTrades: any[] = buyEvents.concat(sellEvents);
-        combinedTrades = combinedTrades.sort(function (a, b) {
-            return b.blockNumber - a.blockNumber;
+        let blockLimit = (this.rootStore.configStore.getDATinfo()).fromBlock;
+        let tradesToReturn = [];
+        
+        if (toBlock > BLOCK_PER_TRADES_FETCH) {
+          blockLimit = toBlock - TRADES_FROM_LAST_BLOCKS;
```
Comment (jpkcambridge) [inline src/stores/datStore.ts:419]: Should be updated to `MAX_BLOCKS_FOR_FETCHING_TRADES`

**Comment by jpkcambridge** [comment]
@AugustoL A bunch of uses of the constants which have updated names also need to be updated.

File: src/stores/datStore.ts [hunk 5]
```diff
@@ -407,19 +410,48 @@ export default class DatStore {
...
+          console.debug('Getting events between blocks', fromBlock, toBlock, tradesToReturn.length);
+
+          if ((tradesToReturn.length < numToGet) && (fromBlock > blockLimit))
+            if ((fromBlock - BLOCK_PER_TRADES_FETCH) < blockLimit) {
+              fromBlock = blockLimit;
+              toBlock = fromBlock;
+              await getEventsBetweenBlocks();
+            } else {
+              fromBlock = fromBlock - BLOCK_PER_TRADES_FETCH;
+              toBlock = toBlock - BLOCK_PER_TRADES_FETCH;
```
Comment (jpkcambridge) [inline src/stores/datStore.ts:440]: `BLOCK_PER_TRADES_FETCH` needs to be updated to `BLOCKS_PER_TRADES_FETCH` in a few places

**Comment by AugustoL** [comment]
@jpkcambridge good catch, shoudl eb good now.

**Pull request merged**